app/models/tagset_models/owned_tag_set.rb

```diff
@@ -155,8 +155,16 @@ def add_moderator(pseud)
 
   def owner_changes=(pseud_list)
     Pseud.parse_bylines(pseud_list)[:pseuds].each do |pseud|
-      if self.owners.include?(pseud)
-        self.owners -= [pseud] if self.owners.count > 1
+      # We load the underlying TagSetOwnership record here
+      # as if we need to delete an owner later, we cannot do it via `self.owners`
+      # without potentially deleting all owners and leaving the tag set orphaned.
```

Comment:
I would add a little extra context here. Specifically, we can't use `Array#delete` or `array -= [...]` because both will remove ALL instances of the pseud, not just the first one. For example (but there's probably a cleaner way to word it)
```rb
# Find the first ownership record for the pseud. If there is one, we delete it
# but ONLY the first one to avoid removing the only owner who happens to
# be assigned multiple times. Calling `Array#delete` on self.owners will remove
# the pseud anywhere it appears, which isn't desirable.
# (NOTE: this should only impact existing OwnedTagSets as newer ones will
# be properly de-duplicated.)
```